convert2rhel/checks.py

```diff
@@ -449,3 +453,47 @@ def _bad_kernel_substring(kernel_release):
         )
         return True
     return False
+
+
+def check_package_updates():
+    """Ensures that the system packages installed are up-to-date."""
+    logger.task("Prepare: Checking if the packages are up-to-date")
+
+    yum_stdout, _ = run_subprocess("yum check-update -q", print_output=False)
+
+    # There is a whole blank line before outputting the packages to update
+    # and to output the real counting, we just -1 the counting.
+    packages_to_update_count = yum_stdout.count("\n") - 1
+
+    if packages_to_update_count > 0:
+        logger.critical(
+            "The system has %s packages to be updated. "
+            "To proceed with the conversion, update the packages on your system by executing the following step:\n\n"
+            "1. Run: yum update -y\n" % packages_to_update_count
+        )
+    else:
+        logger.info("System is up-to-date.")
+
+
+def check_installed_kernel_version():
+    """Check if the installed kernel is behind or at the same version in yum repos"""
+    logger.task("Prepare: Checking if the installed kernel version is the most recent.")
+
+    # The latest kernel version on repo
+    repoquery_output, _ = run_subprocess('repoquery --qf "%{version}-%{release}.%{arch}" kernel')
```

Comment:
This could show multiple kernel versions, at least does for me on Fedora. Will this happen on CentOS or Oracle Linux?

Reply:
@SpyTec, Could you provide the output for me? I think it may be getting kernels from other repositories... If so, this is something I haven't tested. Trying to do some more examples to catch those kernels. 

I'm thinking that we can use the same approach as here: https://github.com/oamg/convert2rhel/pull/353/files#diff-cf19bc90f65d9e26e1b6fa7dcd84a18dfba097f65706e211a4f4ac71b1b467f6R389

We could include the BUILDTIME variable and sort it out for the newest one, what do you think?

Reply:
Here's output on my Fedora 33 system:

> [pts/15@camelback /srv/convert2rhel/c2r-vagrant]$ repoquery --qf "%{version}-%{release}.%{arch}" kernel 
> Last metadata expiration check: 3:01:15 ago on Thu 04 Nov 2021 08:32:35 AM PDT.
> 5.14.15-100.fc33.x86_64
> 5.8.15-301.fc33.x86_64


I think it might retrieve the latest package version for each repository?  In Fedora there's a repository for the release and separate repositories for updates so the kernel package could appear in all three of those.

I'm not sure if it's the same with rhel or if they simply have a monolithic repo.

Reply:
I see @abadger, gonna research a bit more on the `repoquery` output. I think it's querying on all repos the user may have enabled in his system, so it indeed could be a problem. The easiest way I can think of fixing this is sorting out the latest `kernel` as I did previously in another PR. 

Thanks for your output!